editor/src/core/shared/dom-utils.ts

```diff
@@ -229,17 +229,25 @@ export function getCanvasRectangleFromElement(
   element: HTMLElement,
   canvasScale: number,
   withContent: 'without-content' | 'with-content' | 'only-content',
+  rounding: 'nearest-half' | 'no-rounding',
 ): CanvasRectangle {
   const scale = canvasScale < 1 ? 1 / canvasScale : 1
 
+  const roundingFn =
+    rounding === 'nearest-half'
+      ? roundToNearestHalf
+      : rounding === 'no-rounding'
+      ? identity
+      : assertNever(rounding)
+
```

Comment:
can you rewrite this to not be a nested ternary and use a `switch` instead? (possibly inside a function I guess)

Reply:
makes sense, pushed it in https://github.com/concrete-utopia/utopia/pull/4390/commits/72b72d4a99c08f1ad88b4f3adb1e3952829071de